cmd/tink-cli/cmd/template/create.go

```diff
@@ -69,16 +74,23 @@ func readAll(reader io.Reader) []byte {
 func addFlags() {
 	flags := createCmd.PersistentFlags()
 	flags.StringVarP(&filePath, "path", "p", "", "path to the template file")
-	flags.StringVarP(&templateName, "name", "n", "", "unique name for the template (alphanumeric and case sensitive)")
-	_ = createCmd.MarkPersistentFlagRequired(fName)
 }
 
 func tryParseTemplate(data string) error {
 	tmpl := *tt.New("")
 	if _, err := tmpl.Parse(data); err != nil {
 		return err
 	}
-	return nil
+	var templ TemplateName
+	err := yaml.Unmarshal([]byte(data), &templ)
+	if err != nil {
+		return err
+	}
+	if templ.Name != "" {
+		templateName = templ.Name
+		return nil
+	}
+	return errors.New("Template does not have `name` field which is mandatory")
```

Comment:
Please move all the error messages to a `const` and ensure they all start with a lower case.